sn_networking/src/driver.rs

```diff
@@ -465,13 +486,29 @@ impl NetworkBuilder {
             libp2p::identify::Behaviour::new(cfg)
         };
 
-        #[cfg(not(feature = "tcp"))]
+        // Default quic transport.
+        // cannot be built for wasm32
+        #[cfg(all(not(feature = "websockets"), not(target_arch = "wasm32")))]
         let main_transport = TokioTransport::new(TransportConfig::new(&self.keypair))
             .map(|(peer_id, muxer), _| (peer_id, StreamMuxerBox::new(muxer)))
             .boxed();
 
-        #[cfg(feature = "tcp")]
-        let main_transport = TokioTransport::new(TransportConfig::default())
+        #[cfg(target_arch = "wasm32")]
+        let main_transport = WebSocketTransport::default()
```

Comment:
same idea as before, move to separate file and create a `create_main_transport` function

Reply:
I did look at this, but things got unwieldy fast. 

I'm not against it, but I don't think it's a blocker for this PR.